plugins/woocommerce/src/Blocks/Templates/AbstractPageTemplate.php

```diff
@@ -67,7 +42,7 @@ public static function get_template_title() {
 	 */
 	public function page_template_hierarchy( $templates ) {
 		if ( $this->is_active_template() ) {
-			array_unshift( $templates, $this->get_slug() );
+			array_unshift( $templates, static::SLUG );
```

Comment:
This should be `self::SLUG`, isn't it?

Reply:
Yup, good catch. Fixed in 866c9b913a37e4496d53f9c64eee3d4d248412b0.

Reply:
Ups, sorry. I was too fast answering. `SLUG` is still static to the classes as it's defined as a `const SLUG`, so we need to call it as `static::SLUG` from the parent class.

While we could make it non-static, I would prefer to leave it as static for now. We are accessing template slugs from several places of the codebase where we don't need to access the entire class, we only need the slug. Keeping it static allows us to reuse the same template SLUG without having to hard-code it and without having to instantiate the entire template class. What do you think?

Reply:
>  I was too fast answering. 

@Aljullu I'm sorry, I was too fast asking too. I see now that you're accessing the constant defined in a child class from the parent class, so we have to use `static::SLUG` in this case.

> While we could make it non-static, I would prefer to leave it as static for now. We are accessing template slugs from several places of the codebase where we don't need to access the entire class, we only need the slug.

As I mentioned above, this comment is purely about syntax. I am not against using a constant for the slug. I think using constant for template slug makes sense, especially for the reasons you mentioned. I'm marking this one as resolved now. Thanks for the explanation!